agent/consul/leader_connect_ca.go

```diff
@@ -1451,6 +1451,22 @@ func (c *CAManager) SignCertificate(csr *x509.CertificateRequest, spiffeID conne
 		defer c.caLeafLimiter.csrConcurrencyLimiter.Release()
 	}
 
```

Comment:
check the limiter behaviour  (consul rate limiter)

Reply:
We are rate limited using `CSRMaxPerSecond` and `CSRMaxConcurrent` even in the case of intermediate expired. That said we also should be safe because I don't see any loop that aggressively retry the leaf generation.